### ISSUE TITLE
Associative array assignment with quoted keys triggers SC2079

### ISSUE THREAD

**Issue by nhed** [koalaman/shellcheck#3005]
#### For bugs
- Rule Id:  SC2079 
- My shellcheck version: 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

readonly -A fname_aa=(
    ["00_00_00_00-foo.yaml"]="1"
    ["00_00_00_01-bar.yaml"]="1"
)

export fname_aa
```

#### Here's what shellcheck currently says:
```
In dummy.bash line 4:
    ["00_00_00_00-foo.yaml"]="1"
     ^--------------------^ SC2079 (error): (( )) doesn't support decimals. Use bc or awk.


In dummy.bash line 5:
    ["00_00_00_01-bar.yaml"]="1"
     ^--------------------^ SC2079 (error): (( )) doesn't support decimals. Use bc or awk.

For more information:
  https://www.shellcheck.net/wiki/SC2079 -- (( )) doesn't support decimals. U...
```

#### Here's what I wanted or expected to see:

I expect to see no complaints!

AFAIK This is a valid initialization of an associative array and I just want my keys quoted (without quotes, single or double the error does not trigger).   I want these quoted with either single or double as it messes my editor up otherwise and general paranoia.

**Comment by nhed** [comment]
Temporary workaround - use `declare` instead of `readonly`

```declare -r -A fname_aa=(...```
